mesa/batchrunner.py

```diff
@@ -67,36 +87,42 @@ def __init__(self, model_cls, parameter_values, iterations=1,
 
         self.display_progress = display_progress
 
+    def _process_parameters(self, params):
+        params = copy.deepcopy(params)
+        bad_names = []
+        for name, values in params.items():
```

Comment:
This typecheck can be enforced with PEP484, will be much simpler. The version requirement will have to be bumped to be at least 3.5.